### ISSUE TITLE
Firefox hang when open `Log In` on www.cisco.com

### ISSUE THREAD

**Issue by Sopor-** [AliasIO/Wappalyzer#2140]
When i open http://www.cisco.com and i then press `Log In` my `Firefox 58.0.2 (64-bit)` will totally stop responding. I can change tabs but if i try to browse it will only start showing that busy line in the tab and nothing more will happening. I can't open any extensions either by clicking on their icons.

![image](https://user-images.githubusercontent.com/5789283/36938424-9c009a44-1f21-11e8-9b8b-68924df68ad3.png)

If i disable Wappalyser 5.4.8 it will work as it should.

**Comment by gadcam** [comment]
Duplicate of #2132 should be fixed by 74bb8fa.

**Comment by gadcam** [comment]
I apologize for my quick reaction : it has not much to do with 74bb8fa in the end.

I guessed that it has nothing to do with asynchronicity and I was right.
After invegistations I finally found out what was going on :
We have two Regexp [Criteo script pattern](https://github.com/AliasIO/Wappalyzer/blob/master/src/apps.json#L2088) and [Carbon Ads script pattern](https://github.com/AliasIO/Wappalyzer/blob/master/src/apps.json#L1523) with which we are experiencing catastrophic backtracking.
We are experiencing something that catastrophic because when we get script we not only get regular scripts links but also inpage scripts.

This script will tell something like "Matching on element 42 took 115663ms.".

```js
const scripts = Array.prototype.slice
    .apply(document.scripts)
    .filter(script => script.src)
    .map(script => script.src);

const scriptsWithoutNoise = scripts.filter(element => element.indexOf("data:text/javascript;") != 0); 

//const regexpCriteo = new RegExp("[^/]*//(?:cas\.criteo\.com|(?:[^/]\.)?criteo\.net)/", "i");
//const regexpCarbonAds = new RegExp("[^\/]*\/\/(?:engine|srv)\.carbonads\.com\/", "i");

const annoyingRegexp = new RegExp("[^/]*//", "i");

function timeRegexp(array){
	var totalTime = 0;
	var i = 0;
	array.forEach(function(element){
		var start = new Date();
		annoyingRegexp.test(element);
		var subTime = new Date() - start;
		console.log("Matching on element " + i + " took " + subTime + "ms.");
		totalTime += subTime;
		i++;
	});
	console.log("Total time: " + totalTime + " ms.");
}

console.log("We will time with all the scripts first :");
timeRegexp(scripts);

console.log("Now we time after filtering the scripts' array");
timeRegexp(scriptsWithoutNoise);
```
